kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProducerMetrics.java

```diff
@@ -89,6 +91,16 @@ final class ProducerMetrics {
   private static final String RECORD_ERROR_RATE_METRIC_DOC =
       "The average per-second number of record sends that resulted in errors.";
 
+  // rate-limiting
+  static final String RECORD_RATE_LIMITED_COUNT_WINDOWED_METRIC_NAME =
+      "record-rate-limited-count-windowed";
+  private static final String RECORD_RATE_LIMITED_COUNT_WINDOWED_METRIC_DOC =
+      "The total number of record sends that resulted in errors in the given window.";
```

Comment:
Description should be something like "The total number of record sends that were rate limited by the produce rate limiters"

Reply:
While it also doesn't exactly capture what's happening, I prefer using _produce requests_ over _record sends_ because when rate-limiting on Kafka REST's side kicks in, we never get to the `Produce.send` part.

Reply:
@dimitarndimitrov, actually this metric is for our record-level rate-limiter. So records is more accurate than requests.